addons/mail/controllers/discuss/channel.py

```diff
@@ -76,6 +80,11 @@ def discuss_channel_mark_as_seen(self, channel_id, last_message_id, allow_older=
 
     @http.route("/discuss/channel/notify_typing", methods=["POST"], type="json", auth="public")
     def discuss_channel_notify_typing(self, channel_id, is_typing):
+        channel = request.env["discuss.channel"].search([("id", "=", channel_id)])
+        if not channel.group_public_id:
+            public_channel_member = request.env["discuss.channel.member"]._get_as_sudo_from_request(request, channel_id)
+            if not public_channel_member:
+                channel.add_members(request.env.user.partner_id.id)
         channel_member_sudo = request.env["discuss.channel.member"]._get_as_sudo_from_request_or_raise(
```

Comment:
It is inefficient to find the same member twice.
So, what you could do is that update `add_members` to return the newly added members.